doc/poe/poe_hld.md

```diff
@@ -0,0 +1,370 @@
+| PSE HW Version  | SAI_POE_PSE_ATTR_PSE_HARDWARE_VERSION |
+| PSE Temperature | SAI_POE_PSE_ATTR_TEMPERATURE          |
+| PSE Status      | SAI_POE_PSE_ATTR_PSE_STATUS           |
+
+#### Port Capabilities
+
+| Capabilities       | SAI attributes                        |
+| ------------------ | ------------------------------------- |
+| Front Panel ID     | SAI_POE_PORT_ATTR_FRONT_PANEL_ID      |
+| Standard           | SAI_POE_PORT_ATTR_STANDARD            |
+| Admin Enable State | SAI_POE_PORT_ATTR_ADMIN_ENABLED_STATE |
+| Power Limit        | SAI_POE_PORT_ATTR_POWER_LIMIT         |
+| Power Priority     | SAI_POE_PORT_ATTR_POWER_PRIORITY      |
+| Power Consumption  | SAI_POE_PORT_ATTR_CONSUMPTION         |
+| Status             | SAI_POE_PORT_ATTR_STATUS              |
+| Detailed Status    | SAI_POE_PORT_ATTR_DETAILED_STATUS     |
+
+
+### Configuration and management 
+
+#### Manifest (if the feature is an Application Extension)
+Feature is not an Application Extension
+
+#### CLI
+
+##### Show commands
+- show poe status
+
+```
+PoE system information
+================================
+Status            : ready
+Total PoE Ports   : 48
+
+Total Power       : 714.500 W
+Power Consumption : 12.500 W
+Power Available   : 702.000 W
+
+ ```
+
+- show poe interface \<ifname\>
+```
+
+Port         Status      En/Dis  Priority Protocol      Class  PWR Consump  PWR Limit   Voltage   Current
+------------ ----------- ------- -------- ------------- ----- ------------ ---------- --------- ---------
+Ethernet0    delivering  enable  high     802.3BT/High      4      6.100 W   80.600 W  54.000 V   0.113 A
+Ethernet1    delivering  enable  crit     802.3BT/High      4      6.500 W   80.600 W  54.000 V   0.120 A
+Ethernet2    searching   enable  low      802.3BT/High      0      0.000 W   80.600 W   0.000 V   0.000 A
+Ethernet3    searching   enable  crit     802.3BT/High      0      0.000 W   80.600 W   0.000 V   0.000 A
+```
+
+##### Config commands
+
+- config poe interface status \<ifname\> {enable | disable}
+- config poe interface priority \<ifname\> {crit | high | low}
+- config poe interface power_limit \<ifname\> {power_limit}
+
+Examples:
+```
+$ config poe interface status Ethernet0 enable
+$ config poe interface priority Ethernet0 high
+$ config poe interface power_limit Ethernet0 20.4
+```
+
+**TODO**: update CLI reference https://github.com/sonic-net/sonic-utilities/blob/master/doc/Command-Reference.md
+
+#### YANG model Enhancements
+
+TBD
+
+#### Config DB Enhancements  
+
+New table has been added to the config database to store related PoE configuration parameters:
+
+##### POE_PORT_TABLE
+~~~
+    ;Stores PoE port configuration
+    key         = POE_PORT|ifname         ; ifname with prefix POE_PORT
+    ; field     = value
+    enabled     = "enable" / "disable"    ; enable/disable PoE on port, default "disable"
+    power_limit = 1*3.1DIGIT              ; power limit on PoE port
```

Comment:
Please confirm whether power budget will get updated when the PSU gets removed from the switch. Whether this will be taken care by the hardware ?

Reply:
Typically it is done by the HW, from the SW point of view we just need to initialize the power banks.